app/scripts/configImporter.js

```diff
@@ -390,7 +399,9 @@
                             }]);
                         return requests;
                     }, []);
-                return multiCall(server, requests, credentials).then(function(res) {
+                // return multiCall(server, requests, credentials)
+                return processUserImports(server,requests, credentials)
```

Comment:
Why you need to replace multicall with your method? It won't work because you don't wait for results in it and return requests (list of calls). So you'll call updateImported data with incorrect parameteers on next step.

Reply:
Multicall returns the index of failed request

Reply:
Nick I fought with this for days and eventually this strategy produced the results we required. Can we use it as it is now. It works. And optimise later please?

Reply:
At first. You don't process results. In the next step you should update userId for each entity that depends on it.
Of course your method will work very slow. What if you try to import 1000 users? You'll send 1000 requests to the server. It will just block you.

Reply:
Please also remove ```case user``` block in ```generateAddRequests``` method

Reply:
Hi Nick, the multicall fails and I was not able to find the reason why. I have just tested it and it does not import all users. My procedure seems to import all users as expected, this has been tested by our team and they have found no problems with my version. Yes I agree it is not ideal and many users will cause problems/be slow. Please authorise this update and I will work on figuring out why the multicall is not working as expected in the interim and post that change on the next release. I will inform the users of the situation. ok?

Reply:
No. I can't accept this change. It's not only performance problems. Your code doesn't process user import results. All entities that are related to the users won't be added or will be set up incorrectly.

Reply:
As I mentioned above you can easily debug multicall. Just take a look at the response. There is failed request index in it and error message. You can find the corresponding request and debug it manually.

Reply:
changes reverted.